### PR TITLE
fix: import cache

### PR DESCRIPTION
**Pull request by PengBoUESTC**
#### `Commit Message Summary (CHANGELOG)`

```
add query for import file path, to avoid import cache.
```

### `Type`

- [ ] CI
- [x] Fix
- [ ] Perf
- [ ] Docs
- [ ] Test
- [ ] Chore
- [ ] Style
- [ ] Build
- [ ] Feature
- [ ] Refactor

### `SemVer`

- [x] Fix (:label: Patch)
- [ ] Feature (:label: Minor)
- [ ] Breaking Change (:label: Major)

### `Issues`

https://github.com/vitejs/vite/issues/15745

- Fixes `#1`

### `Checklist`

- [ ] Lint and unit tests pass with my changes
- [ ] I have added tests that prove my fix is effective/works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes are merged and published in downstream modules

### CODE REVIEW

**Review by brc-dd — APPROVED** [review]

**Comment by brc-dd** [comment]
Yeah. It's quite standard way to avoid caching. For most cases, it won't affect anything (as generally every tool calls postcss-load-config once or twice). But for test runners that don't properly isolate stuff, this is needed if you're writing/reading config from same paths. Jiti supports `JITI_CACHE` env variable to customize this behavior, but for native imports, I guess it's fine to bypass caching.

**Comment by benjaminpreiss** [comment]
> There is a memory leak associated with this though (refer comments in [nodejs/node#49442](https://github.com/nodejs/node/issues/49442)). But those are hard to avoid, and like I said since tools don't load postcss config too many times, it should fine.

I'd like to understand this better (as I have a usecase where the postcss.config.js would reload quite often)

**Comment by brc-dd** [comment]
> I'd like to understand this better

From what I understand, node's module cache has circular references and gc cannot exactly know if a particular module is no longer used.

With this PR, each call to postcss-load-config will make node treat the config as a separate module, and the module cache will grow over time. This behavior is same on deno.

Without this PR, each call to postcss-load-config will return the same config even if the config has changed. Bun ignores the query parameters, so this PR will have no effect there.

**Comment by benjaminpreiss** [comment]
> With this PR, each call to postcss-load-config will make node treat the config as a separate module, and the module cache will grow over time. This behavior is same on deno.

I see - Is there no garbage collection active on unused modules in the module cache?

**Comment by brc-dd** [comment]
Astro has this - https://github.com/withastro/astro/blob/b956149b2805fdd977d20e267f05c6c4872e2704/packages/astro/src/core/config/vite-load.ts#L45

Nx has - https://github.com/nrwl/nx/blob/09abd39ecc1e11ef94b404d4512fca3526d20d31/packages/remix/src/utils/remix-config.ts#L52

Sindre's repo has this way mentioned - https://github.com/sindresorhus/import-fresh?tab=readme-ov-file#esm

File: src/req.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ async function req(name, rootFile = __filename) {
   let url = createRequire(rootFile).resolve(name)
 
   try {
-    return (await import(pathToFileURL(url).href)).default
+    return (await import(`${pathToFileURL(url).href}?t=${Date.now()}`)).default
```
Comment (brc-dd) [inline src/req.js:19]: .href can be removed, template literal will call toString by default which returns href

Comment (PengBoUESTC) [reply src/req.js]: Done

**Comment by ai** [comment]
Thanks. Released in 5.0.3.

**Comment by PengBoUESTC** [comment]
> Thanks. Released in 5.0.3.

❤️

**Pull request merged**